### PR TITLE
RD-5041 Deny modification of  parent_id  and consumer_id labels

### PR DESCRIPTION
**Pull request by leon-good-life**
## Description
Jira ticket: RD-5041

- [x] In “Add labels“ modal, Disable Plus button for csys-obj-parent and csys-consumer-id in adding labels modal
- [x] In “Add labels“ modal, update popup message with a description
- [x] In labels table (in Labels widget) - disable edit and remove buttons for keys csys-consumer-id and csys-obj-parent
- [x] In “Manage labels” modal (Deployment Actions button in Deployments widget or Deployments View widget) - block adding/removing labels with csys-consumer-id and csys-obj-parent keys; add tooltip when hovering over disabled buttons with a description. (**Development note:** I tested it, this point is automatically done because of the shared code.)

## Screenshots / Videos

Add Labels Modal - before the changes:
Screen Shot 2022-08-13 at 14 06 42

Add Labels Modal - after the changes:
Screen Shot 2022-08-13 at 14 03 51

Labels Table - before the changes:
Screen Shot 2022-08-13 at 14 10 15

Labels Table - after the changes:
Screen Shot 2022-08-13 at 14 12 32

## Checklist

- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I verified that all tests and checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests
`deployment_actions_spec` 14 aug 2022 passed:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/3444/pipeline

running all system tests:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/3443/pipeline

 

## Documentation
https://github.com/cloudify-cosmo/docs.getcloudify.org/pull/2296

### CODE REVIEW

File: widgets/common/src/labels/LabelsInput.tsx [hunk 1]
```diff
@@ -73,8 +73,13 @@ const LabelsInput: FunctionComponent<LabelsInputProps> = ({
         return !!_.find(allLabels, newLabel);
     })();
     const newLabelKeyIsNotPermitted = newLabelKey.startsWith(internalKeyPrefix) && !reservedKeys.includes(newLabelKey);
+    const newLabelShouldBeDenied = newLabelKey === 'csys-obj-parent' || newLabelKey === 'csys-consumer-id';
```
Comment (leon-good-life) [inline widgets/common/src/labels/LabelsInput.tsx:76]: `reservedKeys` is the list consist of: csys-consumer-id, csys-obj-parent, csys-location-lat, csys-env-type, csys-environment, csys-obj-name, csys-location-name, csys-location-long, csys-wrcp-services, csys-wrcp-group-id, csys-obj-type.

According to this [Jira comment](https://cloudifysource.atlassian.net/browse/RD-5041?focusedCommentId=77263), only `csy-consumer-id` and `csys-obj-parent` should be denied and for other labels that begin with csys- I only need to show the tooltip without disabling Add/Edit/Delete.

**Comment by leon-good-life** [comment]
`deployment_actions_spec` 2022 passed:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/3444/pipeline

File: widgets/common/src/labels/LabelsInput.tsx [hunk 2]
```diff
@@ -30,7 +30,9 @@ function useReservedKeys(toolbox: Stage.Types.Toolbox) {
         setFetchingReservedKeys();
         actions
             .doGetReservedLabelKeys()
-            .then(setReservedKeys)
+            .then(keys => {
+                setReservedKeys(keys.filter((key: string) => key !== 'csys-consumer-id' && key !== 'csys-parent-id'));
```
Comment (kubama) [inline widgets/common/src/labels/LabelsInput.tsx:34]: In other places you are using `csys-obj-parent` but here `csys-parent-id` - is this correct?

**Review by kubama — COMMENTED** [review]
I can see label keys being used in three different files. It would be good to extract them into common collection. Or to provide a common function like `isLabelModifiable` or something like this.

File: widgets/common/src/labels/LabelsInput.tsx (hunk 2 above)
Comment (leon-good-life) [reply widgets/common/src/labels/LabelsInput.tsx:34]: it is a mistake, it needs to be `csys-obj-parent`

Comment (leon-good-life) [reply widgets/common/src/labels/LabelsInput.tsx]: I addressed it: [db441c0](https://github.com/cloudify-cosmo/cloudify-stage/pull/2130/commits/db441c014b3f9c8d8ae405de9b2af14e0efa0a67)

**Comment by leon-good-life** [comment]
I started `deployment_actions_spec` on Jenkins: https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/3450/pipeline

**Comment by leon-good-life** [comment]
the test passed

File: test/cypress/integration/widgets/deployment_actions_spec.ts [hunk 3]
```diff
@@ -1,5 +1,9 @@
 import type { LabelInputType } from '../../../../widgets/common/src/labels/types';
 
+function isLabelModifiable(key: string) {
+    return key !== 'csys-consumer-id' && key !== 'csys-obj-parent';
+}
```
Comment (kubama) [inline test/cypress/integration/widgets/deployment_actions_spec.ts:5]: We already have this defined in `widgets/common/src/labels/common.ts`

Comment (leon-good-life) [reply test/cypress/integration/widgets/deployment_actions_spec.ts]: I addressed it

File: widgets/labels/src/LabelsTable.tsx [hunk 4]
```diff
@@ -4,6 +4,7 @@ import { useState, useEffect } from 'react';
 import LabelValueInput from './LabelValueInput';
 import DeleteConfirmModal from './DeleteConfirmModal';
 import AddLabelsModal from './AddLabelsModal';
+import { isLabelModifiable } from '../../common/src/labels/common';
```
Comment (kubama) [inline widgets/labels/src/LabelsTable.tsx:7]: The correct way for referencing this would be through Stage.Common

Comment (leon-good-life) [reply widgets/labels/src/LabelsTable.tsx]: I addressed it

Comment (leon-good-life) [reply widgets/labels/src/LabelsTable.tsx]: bdb5e7cae07a2c4a60a2c516367201b7b079375d

**Comment by leon-good-life** [comment]
I started `deployment_actions_spec`: https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/3457/pipeline

**Comment by leon-good-life** [comment]
The test passed

**Review by kubama — APPROVED** [review]

**Pull request merged**